invenio_users_resources/services/users/service.py

```diff
@@ -34,6 +42,74 @@ def user_cls(self):
         """Alias for record_cls."""
         return self.record_cls
 
+    @unit_of_work()
+    def create_via_admin(self, identity, data, raise_errors=True, uow=None):
+        """Create a user."""
+        self.require_permission(identity, "create")
+        # Remove the following from data dict as will fail schema validation.
+        for key in [
+            "id",
+            "domain",
+            "preferences",
+            "profile",
+            "identities",
+            "domaininfo",
+        ]:
+            if key in data:
+                data.pop(key)
+        data, errors = self.schema.load(
+            data,
+            context={"identity": identity},
+            raise_errors=raise_errors,
+        )
+        # create flask user
+        user, errors = self._create_user_as_admin(data)
+        # run components
+        self.run_components(
+            "create",
+            identity,
+            data=data,
+            user=user,
+            errors=errors,
+            uow=uow,
+        )
+        # get email token and reset info
+        account_user = current_datastore.get_user(user.id)
+        token, reset_link = default_reset_password_link_func(account_user)
+        # trigger celery task to send email.
+        uow.register(
+            TaskOp(
+                execute_reset_password_email,
+                user_id=user.id,
+                token=token,
+                reset_link=reset_link,
+            )
+        )
+        return self.result_item(
+            self, identity, user, links_tpl=self.links_item_tpl, errors=errors
+        )
+
+    def _create_user_as_admin(
+        self,
+        user_info: dict,
+    ):
+        """Create a new user with auto-generated password."""
+        # Generate password and add to user_info dict
+        generated_password = "".join(
+            random.choices(string.ascii_letters + string.digits, k=12)
```

Comment:
We should use the [secrets](https://docs.python.org/3/library/secrets.html) module instead of random, which is not very important since the password will never be effectively used.